naturalspeech2_pytorch/naturalspeech2_pytorch.py

```diff
@@ -1367,8 +1402,11 @@ def forward(
             x_start = alpha * audio - sigma * pred
 
         _, ce_loss = self.codec.rq(x_start, codes)
-
-        return loss + self.rvq_cross_entropy_loss_weight * ce_loss
+        # pitch and duration loss
+        duration_loss = F.l1_loss(duration, duration_pred) if exists(duration) else duration_pred
+        pitch_loss = F.l1_loss(pitch, pitch_pred) if exists(pitch) else pitch_pred
+        
+        return loss + self.rvq_cross_entropy_loss_weight * ce_loss + duration_loss + pitch_loss
```

Comment:
if the `duration` and `pitch` targets were not given above, i'm not sure it can be summed to the scalar loss

Reply:
pitch is passed in forward and gt duration will be used from aligner aln_hard , updating gt duration --> aln_hard

Reply:
@manmay-nakhashi yup, but should still handle the case where it is not passed in correctly `pitch_loss = F.l1_loss(pitch, pitch_pred) if exists(pitch) else pitch_pred # pitch_pred is not a loss`

Reply:
i think its fine just to rewrite it to `pitch_loss = F.l1_loss(pitch, pitch_pred) if exists(pitch) else 0.`